### DISCUSSION TITLE
Building message notifications

### DISCUSSION THREAD

**Discussion by jtgi** [MetaMask/snaps#915] [Q&A] [answered]
gm from ethsf,

We are interested in using a snap to show a notification in metamask when their account has received a message.
The messaging api is a third party off-chain service.

The stumbling point at the moment is: how to do some kind of polling or periodic checking for messages (ie: an api call) within a snap, without requiring clients to implement new rpc methods. Forcing clients to support our snaps rpc methods would kill chances of adoption.

Ideally, I could:
1. Run a background service and poll an api and trigger a snap notification whenever, or
2. If no background service, hook into common rpc events, like `eth_requestAccounts` and trigger a fetch for new messages to then trigger a notification.

I've looked at the docs closely and it seems like this isn't possible, is there any workaround–even if hacky?
Could I hook into transaction insights and trigger a notification?

Thank you!

**Comment by ritave** [comment, marked as answer]
Hello @jtgi, last week we've build [cronjobs](https://github.com/MetaMask/snaps-monorepo/pull/651), which allow you to run background periodic snaps.

It's not currently released inside Flask yet, so please watch for upcoming [announcements](https://github.com/MetaMask/snaps-monorepo/discussions/categories/announcements).

**Reply by jtgi** [reply]
This is just what I would need. Thank you for your response!

**Comment by Navaneeth-Bysani** [comment]
Hi @ritave Is there a way to do this notification kind of service without polling? Like using websockets or something!

**Reply by ritave** [reply]
We have some floating ideas to do push-based APIs in the future, but nothing concrete as of right now.

**Reply by Navaneeth-Bysani** [reply]
@ritave  In continuation to the previous one, is it a good idea to use [push (EPNS)](https://push.org/) to implement notifications? Doing this would require a central server that sends a notification to the user. The server doesn't need to store data, so practically, there is no leveraging of user data (which is the principle of web3)

**Reply by ritave** [reply]
@Navaneeth-Bysani we're aware about EPNS, we didn't have the time to look at any concrete solution yet.
